bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -230,17 +232,19 @@ protected void removeModuleHandlerFactory(ModuleHandlerFactory moduleHandlerFact
      */
     @Override
     public Rule add(Rule rule) {
-        if (rule == null) {
-            throw new IllegalArgumentException("The added rule must not be null!");
-        }
         String rUID = rule.getUID();
         if (rUID == null) {
             rUID = ruleEngine.getUniqueId();
             super.add(initRuleId(rUID, rule));
         } else {
             super.add(rule);
         }
-        return get(rUID);
+        Rule ruleCopy = get(rUID);
```

Comment:
If they are helpful as they remove a warning at another place as in this case, what speaks against it?